### PR TITLE
Dojo 2 Implementation

### PR DESCRIPTION
**Pull request by agubler**
Dojo 2 implementation of the Hacker News PWA

**Github Repo:** https://github.com/agubler/dojo-2-hn
**Running Application:** https://dojo-2-hnpwa-d668d.firebaseapp.com/

(Please let me know what the weights needed are and I'll update the references)

Thanks!

### CODE REVIEW

**Comment by agubler** [comment]
Hey, are new implementations still being accepted?

**Comment by addyosmani** [comment]
@agubler Absolutely. Sorry for the crazy delays on reviews. We're a little behind due to releases and events related to other projects.

**Comment by addyosmani** [comment]
UX check:

1. Navigated to all top-level routes and they appear to work as expected
2. Child routes like comments are all working as expected
3. Content is rendering while offline as expected
4. I ran into some network connection issues/occasional breakage with the new view. This was because the app displays the next link even when no further content is available. https://agubler.github.io/dojo-2-hn/dist/#jobs/1 should be a repro you can try out of this.

image

5. Navigating back and forth between offline and online states appears to work as expected. 
6. Your UX is pretty close to the PreactHN implementation which I usually use as reference.

**Comment by agubler** [comment]
@addyosmani completely understand! Thanks for taking the time out to review the implementation.

I've pushed some changes to deal with the error next page link when no more items exists, any further feedback would be greatly appreciated 😄

**Comment by anubhav7495** [comment]
@agubler `Ask HN` links are broken. They should open the comments view but lead to a 404 page right now.

**Comment by agubler** [comment]
@anubhav7495 thanks for pointing that out, I didn’t know that they needed to open in the comments view. I’ll fix that up. Thanks again.

**Comment by agubler** [comment]
@anubhav7495 The `Ask HN` links should now be fixed and route to the comments view. Thanks.

**Comment by agubler** [comment]
@addyosmani Have mades some updates to the implementation, hopefully ready for another review. Thank you in advance.

**Comment by agubler** [comment]
Hey, I know everyone is mega busy but it’d be  great to know what else I need to do to get this mergeable. Thanks in advance!

**Comment by agubler** [comment]
Hey @addyosmani, sorry to pester but do you think anyone will get a chance to give this a re-review soon? It would be great to get the implementation approved and merged 😄

**Comment by anubhav7495** [comment]
@agubler I will take a look at it tomorrow first things. sorry for the delay. Thanks.

**Comment by anubhav7495** [comment]
LGTM. @addyosmani

**Comment by agubler** [comment]
Thank you @anubhav7495!

File: site/_apps/dojo-2.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+authors:
+    - name: 'Anthony'
+lighthouse: '91/100'
+interactive-em: '5s'
+interactive-faster-3g: '3.9s'
+lighthouse-link: https://www.webpagetest.org/lighthouse.php?test=170928_M7_f23cff7be214c26bb1ca5a559bfa5200&run=2
+wpt-em-link: https://www.webpagetest.org/result/170928_54_819799e9cf8b69190819bc5bde259607/
+wpt-faster-3g-link: https://www.webpagetest.org/result/170928_M7_f23cff7be214c26bb1ca5a559bfa5200/
+image: /assets/images/dojo-2-hnpwa.png
+app-link: https://agubler.github.io/dojo-2-hn/dist/#top/1
```
Comment (addyosmani) [inline site/_apps/dojo-2.txt]: After clearing out my Cache API caches, navigating to https://agubler.github.io/dojo-2-hn/dist/#top/1 doesn't appear to fetch any stories. It renders the following:

image

We should probably update the app-link to https://agubler.github.io/dojo-2-hn/dist/ which does seem to work correctly.

**Pull request merged**